pkg/source/resource_test.go

```diff
@@ -0,0 +1,188 @@
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest/fake"
+	"k8s.io/client-go/restmapper"
+	k8sscheme "k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/google/go-cmp/cmp"
+
+	cartov1alpha1 "github.com/vmware-tanzu/apps-cli-plugin/pkg/apis/cartographer/v1alpha1"
+	clitesting "github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime/testing"
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/source"
+)
+
+func TestWorkloadOptionsFetchResourceObjects(t *testing.T) {
+	defaultNamespace := "default"
+	workloadName := "my-workload"
+	pod1Die := diecorev1.PodBlank.
+		MetadataDie(func(d *diemetav1.ObjectMetaDie) {
+			d.Name("pod1")
+			d.Namespace(defaultNamespace)
+			d.AddLabel(cartov1alpha1.WorkloadLabelName, workloadName)
+		}).Kind("pod")
+
+	scheme := runtime.NewScheme()
+	fakeClient := clitesting.NewFakeCliClient(clitesting.NewFakeClient(scheme))
+	table := &metav1.Table{}
+	tests := []struct {
+		name        string
+		args        []string
+		shouldError bool
+		expected    []metav1.TableRow
+		builder     *resource.Builder
+	}{
+		{
+			name:        "Fetch Resource Object successfully",
+			args:        []string{"pods"},
+			shouldError: false,
+			builder: resource.NewFakeBuilder(
+				func(version schema.GroupVersion) (resource.RESTClient, error) {
+					codec := k8sscheme.Codecs.LegacyCodec(scheme.PrioritizedVersionsAllGroups()...)
+					UnstructuredClient := &fake.RESTClient{
+						NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
+						Resp:                 &http.Response{StatusCode: http.StatusOK, Header: clitesting.DefaultHeader(), Body: clitesting.PodV1TableObjBody(codec, []client.Object{pod1Die})},
+					}
+					return UnstructuredClient, nil
+				},
+				fakeClient.ToRESTMapper,
+				func() (restmapper.CategoryExpander, error) {
+					return resource.FakeCategoryExpander, nil
+				},
+			),
+			expected: append(table.Rows, metav1.TableRow{
+				Cells: []interface{}{"pod1", "0/0", "", int64(0), "<unknown>", "<none>", "<none>", "<none>", "<none>"},
+			})},
+		{
+			name:        "Fetch Empty Resource Object successfully",
+			args:        []string{"pods"},
+			shouldError: false,
+			builder: resource.NewFakeBuilder(
+				func(version schema.GroupVersion) (resource.RESTClient, error) {
+					codec := k8sscheme.Codecs.LegacyCodec(scheme.PrioritizedVersionsAllGroups()...)
+					UnstructuredClient := &fake.RESTClient{
+						NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
+						Resp:                 &http.Response{StatusCode: http.StatusOK, Header: clitesting.DefaultHeader(), Body: clitesting.PodV1TableObjBody(codec, []client.Object{})},
+					}
+					return UnstructuredClient, nil
+				},
+				fakeClient.ToRESTMapper,
+				func() (restmapper.CategoryExpander, error) {
+					return resource.FakeCategoryExpander, nil
+				},
+			),
+		},
+		{
```

Comment:
```suggestion
		},{
```

Reply:
Do the same for all tests. This is the pattern we have followed on other tests. easier to read and reduce lines

Reply:
Updated